### DISCUSSION TITLE
Show function name in each log message

### DISCUSSION THREAD

**Discussion by kaihendry** [KyleRoss/node-lambda-log#56] [Q&A] [answered]
How would I go about showing the function name AWS_LAMBDA_FUNCTION_NAME in each log message?

**Comment by KyleRoss** [comment, marked as answer]
You can add it as part of your global metadata. Here's an example:

```js
const log = require('lambda-log');

exports.handler = async function(event, context) {
	log.options.meta = {
		functionName: context.functionName
	};

	log.info('Test log');
	// => { "level": "info", "msg": "Test log", "functionName": "my-lambda-function", "_tags": [] }
};
```

Setting it within the global metadata will add the function name to every log automatically!

**Comment by kaihendry** [comment]
Thank you! Actually I'm doing something like:


    exports.handler = async function (event, context) {
    log.options.meta.context = context;
    log.debug('event:', event);


But the context object is a bit heavy weight!!
